2-Draft-Accepted/RFCNNNN-TimeSpan-Literals.md

```diff
@@ -0,0 +1,140 @@
+---
+RFC: RFCNNNN
+Author: Anselm Schüler
+Status: Draft
+Area: Syntax and Cmdlet implementation
+Plan to implement: Yes
+Breaking change: Yes
+---
+
+# `TimeSpan` Literals and support for `TimeSpan` objects in Cmdlets such as Start-Sleep
+
+A new syntax expansion that adds support for `TimeSpan` literals, and cmdlets such as `Start-Sleep` accepting `TimeSpan` objects by default - this entails adding new suffixes to numeric literals as [described here](https://docs.microsoft.com/en-us/powershell/module/Microsoft.PowerShell.Core/About/about_numeric_literals).  
+This is helpful as current methods for time specification have significant drawbacks (see Motivation).  
+This solution is significantly more concise than using `New-TimeSpan` (see User Experience).
+
+Initially, this will be implemented in an experimental feature.
+
+## Motivation
+
+### Practicality
+
+Cmdlets such as `Start-Sleep` offer both `-Seconds` and `-Milliseconds` as parameters.
+This style of time input is functional, but has a few issues:
+
+- Expanding the number of units requires updating the argument handling and the function body itself
+- Implementation in custom scripts or cmdlets may require boilerplate
+- Available options may not be consistent across cmdlets
+
+### Semantics
+
+Commands such as `New-TimeSpan`, which is the premier way to create timespan objects, are usually used in a manner where the returned object is a persistent object that later gets mutated or is itself effectful.  
+By using a suffix, the intent is made clearer by the more functional style.
+
+## User Experience
+
+Example of expressions and their equivalents in current PowerShell:
+
+| New syntax | With `New-TimeSpan` | With type conversion |
+| --- | --- | --- |
+| `10sec` | `New-TimeSpan -Seconds 10` | `[TimeSpan]100000000` |
+| `15min` | `New-TimeSpan -Minutes 15` | `[TimeSpan]9000000000` |
+| `0xFFyear` | `New-TimeSpan -Days (0xFF * 365)` | `[TimeSpan]80416800000000000` |
+
+### Examples of user interaction
+
+```pwsh
+Start-Sleep 20sec
```

Comment:
Can you include an example of something like 1 min and 30 secs?  `Start-Sleep (1min + 30sec)`

Reply:
I have!